src/components/organisms/AdminProfileModal/AdminProfileModal.tsx

```diff
@@ -0,0 +1,136 @@
+import { Fragment, useCallback, useRef } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+
+import { REACT_BOOTSTRAP_MODAL_HIDE_DURATION } from "settings";
+
+import { useProfileModalControls } from "hooks/useProfileModalControls";
+import { useShowHide } from "hooks/useShowHide";
+
+export const AdminProfileModal = () => {
+  const { hasSelectedProfile, closeUserProfileModal } =
+    useProfileModalControls();
+
+  const {
+    isShown: isModalShown,
+    hide: hideModal,
+    show: showModal,
+  } = useShowHide(true);
+
+  const hideHandler = useCallback(async () => {
+    hideModal();
+    // when closeUserProfileModal is called modal hide animation starts
+    // but selectedUserId is immediately undefined and while the modal is sliding away
+    // an error message from ProfileModalFetchUser is shown
+
+    // This is to fix that.
+    window.setTimeout(() => {
+      closeUserProfileModal();
+      showModal();
+    }, REACT_BOOTSTRAP_MODAL_HIDE_DURATION);
+  }, [closeUserProfileModal, hideModal, showModal]);
```

Comment:
I would have thought we could use Headless UI's Transition to avoid this. Is there something I'm not understanding here?

Reply:
I've just copied that from the previous ProfileModal component.
I'll check the Headless UI's Transition tomorrow morning.

Reply:
removed that handler: https://github.com/sparkletown/sparkle/pull/2812/commits/9f0e19504410650d3f6936c81dfbd764241fc028